builder/amazon/common/cli_config.go

```diff
@@ -0,0 +1,95 @@
+package common
+
+import (
+	"fmt"
+    "os"
+    "path"
+
+	"github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/credentials"
+    "github.com/aws/aws-sdk-go/service/sts"
+    "github.com/go-ini/ini"
+)
+
+type CLIConfig struct {
+    SourceProfile   string
+
+    Source          credentials.Value
+    AssumeRoleInput sts.AssumeRoleInput
+}
+
+// Sets params in the struct based on the file section
+func (c *CLIConfig) Prepare(name string) (error) {
```

Comment:
I think this API should be something like: `NewFromProfile(name string) (*CLIConfig, error)`

Assume that we can provide all of the required information up-front and either error (something was missing or invalid configuration was provided) or we can return the struct all ready to go.

Also, I'm curious whether we need this intermediate struct or whether we can just return `*credentials.Credentials` outright, which I think simplifies the internal logic a bit.